tests/user-registration.spec.js

```diff
@@ -0,0 +1,504 @@
+            password: 'Secret10',
+            repeatPassword: 'Secret10'
+        }
+        await expect(signUpPopup.getRegisterBtn).toBeDisabled()
+        await signUpPopup.fillRegistrationForm(userData)
+        await expect(signUpPopup.getRegisterBtn).toBeEnabled()
+        await signUpPopup.saveNewUser()
+        await expect(page).toHaveURL(garagePage.getGaragePageUrl)
+    })
+
+    test('Check that user can be registered with maximal allowed name, last name and password length', async ({page})=> {
+        const signUpPopup = new SignUpPopup(page)
+        const garagePage = new GaragePage(page)
+        const userData = {
+            name: 'somelongnamefortests',
+            lastName: 'somelongnamefortests',
+            email: 'aqa-ipod2@test.com',
+            password: 'Secret10Secret1',
+            repeatPassword: 'Secret10Secret1'
+        }
+        await expect(signUpPopup.getRegisterBtn).toBeDisabled()
+        await signUpPopup.fillRegistrationForm(userData)
+        await expect(signUpPopup.getRegisterBtn).toBeEnabled()
+        await signUpPopup.saveNewUser()
+        await expect(page).toHaveURL(garagePage.getGaragePageUrl)
+    })
+
+    test.afterEach(async ({page})=> {
+        const settingsPage = new SettingsPage(page)
+        const removeAccountPopup = new RemoveAccountPopup(page)
+    
+        await settingsPage.visitSettingsPage()
+        await settingsPage.openRemoveAccountPopup()
+        await removeAccountPopup.removeAccount()
+        await expect(page).toHaveURL('')
+    })
+})
+test.describe("Unsuccessful user registration", ()=> {
+
+    test.beforeEach(async ({page})=> {
+        const homePage = new HomePage(page)
+        const signInPopup = new SignInPopup(page)
+        const signUpPopup = new SignUpPopup(page)
+
+        await homePage.visitHomePage()
+        await homePage.openSignInPopup()
+        await signInPopup.openRegistrationPopup()
+        await expect(signUpPopup.getSignUpPopup).toBeVisible()
+    })
+
+    test('Check that name field is required', async ({page})=> {
+        const signUpPopup = new SignUpPopup(page)
+        const userData = {
+            name: '',
+            lastName: 'po',
+            email: 'aqa-ipod@test.com',
+            password: 'Secret10',
+            repeatPassword: 'Secret10'
+        }
+        await expect(signUpPopup.getRegisterBtn).toBeDisabled()
+        await signUpPopup.fillRegistrationForm(userData)
+        await expect(signUpPopup.getRegisterBtn).toBeDisabled()
+        await expect(signUpPopup.getNameInput).toHaveClass(/is-invalid/) //checks that border color is red by using corresponding class
+        await expect(signUpPopup.getLastNameInput).not.toHaveClass(/is-invalid/)
+        await expect(signUpPopup.getEmailInput).not.toHaveClass(/is-invalid/)
+        await expect(signUpPopup.getPasswordInput).not.toHaveClass(/is-invalid/)
+        await expect(signUpPopup.getRepeatPasswordInput).not.toHaveClass(/is-invalid/)
```

Comment:
also check that input with invalid value has red border

Reply:
fixed